src/dswx_sar/pre_processing.py

```diff
@@ -0,0 +1,407 @@
+import os
+import pathlib
+import time
+import numpy as np
+import logging
+import h5py
+import mimetypes
+
+from osgeo import osr, gdal
+from pathlib import Path
+
+from dswx_sar import filter_SAR
+from dswx_sar import dswx_sar_util
+from dswx_sar.dswx_runconfig import _get_parser, RunConfig
+
+logger = logging.getLogger('dswx_s1')
+
+def pol_ratio(array1, array2):
+    '''
+    Compute ratio between two arrays
+    '''
+    array1 = np.asarray(array1, dtype='float32')
+    array2 = np.asarray(array2, dtype='float32')
+
+    return array1 / array2
+
+def pol_coherence(array1, array2, array3):
+    '''
+    Compute polarimetric coherence from two co-pol and one cross-pol
+    '''
+    array1 = np.asarray(array1, dtype='float32') # VVVV
+    array2 = np.asarray(array2, dtype='float32') # VHVH
+    array3 = np.asarray(array3, dtype='complex') # VVVH
+
+    return np.abs(array3 / np.sqrt(array1 * array2))
+
+class AncillaryRelocation:
+    '''
+    Relocate the ancillary file to have same geographic extents and 
+    spacing as the given rtc file.
+    '''
+    def __init__(self, rtc_file_name, scratch_dir):
+        """Initialized AncillaryRelocation Class with rtc_file_name
+
+        Parameters
+        ----------
+        rtc_file_name : str
+            file name of RTC input HDF5
+        scratch_dir : str
+            scratch directory to save the relocated file
+        """
+
+        self.rtc_file_name = rtc_file_name
+        if  h5py.is_hdf5(rtc_file_name):
+            self.xcoord_rtc, self.ycoord_rtc = dswx_sar_util.read_rtc_lat_lon(rtc_file_name)
+            with h5py.File(rtc_file_name, 'r') as f4:
+                self.epsg = int(np.array(f4['/science/LSAR/GCOV/grids/frequencyA/projection']))
```

Comment:
AFAIK, `/science/LSAR/GCOV/grids/frequencyA/projection` seems to be from NISAR. At least I can tell this dataset does not exist in RTC-S1. Can you take a look?